crates/grit-pattern-matcher/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+//! Grit pattern definitions and matching logic
+//!
+//! This crate contains all the pattern definitions that are at the heart of the
+//! GritQL engine. There's the [`pattern::Matcher`] trait that's implemented by
+//! the patterns, which implements the matching logic.
+
+pub mod binding;
+pub mod constant;
+pub mod constants;
+pub mod context;
+pub mod effects;
+pub mod errors;
+pub mod file_owners;
+pub mod intervals;
+pub mod pattern;
+
+pub use grit_util as util;
```

Comment:
@arendjr Let's not do this. I generally prefer to avoid exporting/aliasing crates - consumers should be fine with importing grit_util directly as well.

Reply:
I'll remove it.

I think the main reason why it's easier for consumers is when they need to upgrade. If they use the dependent crate through a re-export, they can treat it as a transitive dependency. So it will be upgraded along with the direct dependency, which will also take care of making sure the version they're using is a compatible one. If they import `grit_util` directly, they are responsible themselves for upgrading it along with `grit_pattern_matcher`. It's usually not a big deal though, since mistakes tend to be pointed out as compile errors :)

Reply:
Yeah, there are pros and cons but since it has utils that consumers will directly use I think they should import it directly.